apps/backend/api/models/ContextWidget.js

```diff
@@ -0,0 +1,343 @@
+const { reorderTree, replaceHomeWidget } = require("./util/contextWidgets")
+
+module.exports = bookshelf.Model.extend({
+  tableName: 'context_widgets',
+  requireFetch: false,
+  hasTimestamps: true,
+
+  initialize() {
+    this.on('fetching', (model, columns, options) => {
+      if (!options.withRelated) {
+        options.withRelated = []
+      }
+      const relationsToLoad = []
+        
+        if (model.get('custom_view_id')) relationsToLoad.push('customView')
+        if (model.get('view_chat_id')) relationsToLoad.push('viewChat')
+        if (model.get('view_post_id')) relationsToLoad.push('viewPost')
+        if (model.get('view_group_id')) relationsToLoad.push('viewGroup')
+        if (model.get('view_user_id')) relationsToLoad.push('viewUser')
+        relationsToLoad.push('children')
+
+        options.withRelated = options.withRelated.concat(relationsToLoad)
+    })
+  },
+
+  // Relationships
+
+
+  children() {
+    return this.hasMany(ContextWidget, 'parent_id')
+  },
+
+  customView() {
+    return this.belongsTo(CustomView, 'custom_view_id')
+  },
+  group() {
+    return this.belongsTo(Group, 'group_id')
+  },
+
+  highlightNumber() {
+    // TODO CONTEXT: these will need to determine a useful number based on the entity the widget is linked to
+    return 3
```

Comment:
These are currently unused but will be the numbers that indicate activity for various views in the menu